### ISSUE TITLE
[Bug] Pasting a file breaks syntax highlighting by changing extension

### ISSUE THREAD

**Issue by lol768** [jkcclemens/bins#2]
```
$ bins -s gist hello.c
https://gist.github.com/215883b109a0047fe07f5ee229de6a51
```

Link: https://gist.github.com/anonymous/215883b109a0047fe07f5ee229de6a51

File is named `hello.c_1` which changes file extension and causes GitHub Gist to treat the file as plain text. Screenshot:

![](https://lol768.com/i/Tof8tVNziUfJog)

**Comment by lol768** [comment]
@jkcclemens Thanks

You might consider changing how the numbers are added. E.g. for two files with names:

* /dir_a/hello.c
* /dir_b/hello.c

I'd expect names of `hello.c` and `hello.1.c`, *not* `hello.c_1` (which breaks syntax highlighting).

Screenshot:

![](https://lol768.com/i/0GKpLZwyCuR_0w)

**Comment by jkcclemens** [comment]
For the record, @lol768's comment above was implemented in da6f7edddb45eda135b354c6108cc27b534a8298.
